tests/tests.go

```diff
@@ -828,14 +827,57 @@ func getAggregatedTaskStatus(issue *api.Issue) (api.TaskStatus, error) {
 			case api.TaskCanceled:
 				return api.TaskCanceled, nil
 			case api.TaskRunning:
-				running = true
+				return api.TaskRunning, nil
 			case api.TaskPending:
-				running = true
+				return api.TaskPending, nil
 			}
 		}
 	}
-	if running {
-		return api.TaskRunning, nil
+	return api.TaskDone, nil
+}
+
+// waitIssueNextTaskWithTaskApproval waits for next task in pipeline to finish and approves it when necessary.
+// nolint
+func (ctl *controller) waitIssueNextTaskWithTaskApproval(id int) (api.TaskStatus, error) {
```

Comment:
This is too similar to `waitIssuePipelineImpl`.
How about refactor `waitIssuePipelineImpl` to accept a new argument `approveOnce bool`?
If `approveOnce`, then we apply the `approveFunc` once and return. Otherwise apply `approveFunc` as many times as possible like the current implementation.

Reply:
We can do it now after we refactor `getAggregatedTaskStatus`!

Reply:
SG.

Reply:
How about remove `waitIssueNextTaskWithTaskApproval` in this PR, which is `nolint` after all. And we are about to refactor `waitIssuePipelineImpl` instead of adding this.